lib/puppet/type/tomcat_install.rb

```diff
@@ -0,0 +1,62 @@
+Puppet::Type.newtype(:tomcat_install) do
+  desc 'Installs the software into the given directory.'
+
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
+  autorequire(:user) do
+    self[:user]
+  end
+
+  autorequire(:group) do
+    self[:group]
+  end
+
+  newparam(:catalina_home, namevar: true) do
+    desc 'The directory of the Tomcat installation.'
+
+    validate do |value|
+      raise ArgumentError, 'Catalina home is required.' if value.nil?
+    end
+  end
+
+  newparam(:version) do
+    desc 'Specifies the owner of the source installation directory.'
+
+    validate do |value|
+      raise ArgumentError, "Tomcat version must be in the form x.y.z, got #{value}" if value !~ %r{\d{1,2}\.\d{1,2}\.\d{1,2}}
+    end
+  end
+
+  newparam(:source_url) do
+    desc 'Specifies the location of the tomcat install file.'
+
+    validate do |value|
+      raise ArgumentError, 'Source must be a valid URL.' if value !~ %r{https?:\/\/[\S]+}
+    end
+  end
+
+  newparam(:user) do
+    desc 'Specifies the owner of the source installation directory.'
+
+    defaultto 'tomcat'
```

Comment:
As per the hidden discussion at https://github.com/puppetlabs/puppetlabs-tomcat/pull/260#discussion_r159721609 I found this info:

> Taking the Tomcat instances at the [Apache Software Foundation] as an example, the standard configuration is to have all Tomcat files owned by root with group Tomcat and whilst owner has read/write privileges, group only has read and world has no permissions. The exceptions are the logs, temp and work directory that are owned by the Tomcat user rather than root. This means that even if an attacker compromises the Tomcat process, they can't change the Tomcat configuration, deploy new web applications or modify existing web applications.

Via https://tomcat.apache.org/tomcat-7.0-doc/security-howto.html and https://tomcat.apache.org/tomcat-9.0-doc/security-howto.html

Reply:
Another similar reference:
> The Tomcat installation directory (sometimes referred to as CATALINA_HOME) should be installed as a user that is different to the one it will be run as. Under Linux, unpacking the Tomcat distribution as root is the simplest method of doing this.
https://confluence.atlassian.com/jira064/tomcat-security-best-practices-720411783.html (And WAR files as well.)

The practices are also followed for other ASF software, such as Apache HTTPD.